core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -83,6 +101,24 @@ public void start() throws Exception {
       mJournalSystem.waitForCatchup();
     }
 
+    // Add a guard for Ratis Server
+    if (mJournalSystem instanceof RaftJournalSystem) {
+      // Create a health check thread
+      mHeartbeatPool = ExecutorServiceFactories.cachedThreadPool("embedded-journal").create();
```

Comment:
This thread has to call a restart on master modules and journal, so it must not be running in the masters or journal!